### PR TITLE
Reduces usage of ENABLE_HTTPS

### PR DESCRIPTION
**Pull request by arunagw**
We can remove the usage of `ENABLE_HTTPS` variable and that can be removed in later commits.

We can just us what rails is generating for url. 
So sending a URL from controller can be http or https based on what is application using/running. 

We can force the https in `production.rb` instead of having this check in before_filter.

### CODE REVIEW

**Comment by gylaz** [comment]
Would you mind adding a commit description of explaining why this is a good idea?

**Comment by arunagw** [comment]
@gylaz Yes. I have updated the PR description.

Card is here https://trello.com/c/3xft1Ylq

File: app/controllers/activations_controller.rb [hunk 1]
```diff
@@ -23,7 +23,11 @@ def check_repo_plan
   end
 
   def activator
-    @activator ||= RepoActivator.new(repo: repo, github_token: github_token)
+    @activator ||= RepoActivator.new(
+      repo: repo,
+      github_token: github_token,
+      builds_url: builds_url
```
Comment (gylaz) [inline app/controllers/activations_controller.rb]: Instead of passing this in, how about just including the route helpers in the `RepoActivator`?

**Comment by arunagw** [comment]
@gylaz I have updated PR. Please have a look.

**Comment by gylaz** [comment]
Looks good!

File: lib/routing.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+module Routing
+  extend ActiveSupport::Concern
+  include Rails.application.routes.url_helpers
+
+  included do
+    def default_url_options
+      ActionMailer::Base.default_url_options
+    end
+  end
+end
```
Comment (gylaz) [inline lib/routing.rb]: I'm interested about this approach. What does it provide over doing `include Rails.application.routes.url_helpers` in `RepoActivator`?

Comment (arunagw) [inline lib/routing.rb]: The only thing we need to do is set `host` for routes as well. 

Basically 

```Rails.application.routes.default_url_options[:host] = ENV['host']```

**Comment by arunagw** [comment]
@gylaz what you think about this PR?

File: lib/routing.rb (hunk 2 above)
Comment (arunagw) [inline lib/routing.rb]: We can set this in all of our environment files or can just read from mailer options like I did in this concern.

Comment (gylaz) [inline lib/routing.rb]: Could we just set it once in `application.rb` for all envs? If possible, that approach (plus mixing in `Rails.application.routes.url_helpers`) seems simpler.

**Comment by gylaz** [comment]
I had one more question regarding the mixin. Make the call on what makes sense. If this works on staging, then lets merge it.

**Comment by arunagw** [comment]
@gylaz I have removed that module for now. Seems like too much as well.

Will check once on staging before merging.

**Pull request merged**